providers/configcat/src/test/java/dev/openfeature/contrib/providers/configcat/ConfigCatProviderTest.java

```diff
@@ -0,0 +1,180 @@
+        assertEquals(false, configCatProvider.getBooleanEvaluation(USERS_FLAG_NAME, false, evaluationContext).getValue());
+        assertEquals(false, client.getBooleanValue(USERS_FLAG_NAME, false, evaluationContext));
+    }
+
+    @SneakyThrows
+    @Test
+    void shouldThrowIfNotInitialized() {
+        ConfigCatProviderConfig configCatProviderConfig = ConfigCatProviderConfig.builder().sdkKey("test").build();
+        ConfigCatProvider tempConfigCatProvider = new ConfigCatProvider(configCatProviderConfig);
+
+        assertThrows(ProviderNotReadyError.class, ()-> tempConfigCatProvider.getBooleanEvaluation("fail_not_initialized", false, new ImmutableContext()));
+
+        OpenFeatureAPI.getInstance().setProviderAndWait("tempConfigCatProvider", tempConfigCatProvider);
+
+        assertThrows(GeneralError.class, ()-> tempConfigCatProvider.initialize(null));
+
+        tempConfigCatProvider.shutdown();
+
+        // ErrorCode.PROVIDER_NOT_READY should be returned when evaluated via the client
+        assertThrows(ProviderNotReadyError.class, ()-> tempConfigCatProvider.getBooleanEvaluation("fail_not_initialized", false, new ImmutableContext()));
+        assertThrows(ProviderNotReadyError.class, ()-> tempConfigCatProvider.getDoubleEvaluation("fail_not_initialized", 0.1, new ImmutableContext()));
+        assertThrows(ProviderNotReadyError.class, ()-> tempConfigCatProvider.getIntegerEvaluation("fail_not_initialized", 3, new ImmutableContext()));
+        assertThrows(ProviderNotReadyError.class, ()-> tempConfigCatProvider.getObjectEvaluation("fail_not_initialized", null, new ImmutableContext()));
+        assertThrows(ProviderNotReadyError.class, ()-> tempConfigCatProvider.getStringEvaluation("fail_not_initialized", "", new ImmutableContext()));
+    }
+
+    @Test
+    void eventsTest() {
+        configCatProvider.emitProviderReady(ProviderEventDetails.builder().build());
+        configCatProvider.emitProviderError(ProviderEventDetails.builder().build());
+        configCatProvider.emitProviderConfigurationChanged(ProviderEventDetails.builder().build());
+    }
+
+    @SneakyThrows
+    @Test
+    void contextTransformTest() {
+        String userId = "a";
+        String email = "a@a.com";
+        String country = "someCountry";
+        String customPropertyValue = "customProperty_value";
+        String customPropertyKey = "customProperty";
+
+        MutableContext evaluationContext = new MutableContext();
+        evaluationContext.setTargetingKey(userId);
+        evaluationContext.add("Country", country);
+        evaluationContext.add("Email", email);
+        evaluationContext.add(customPropertyKey, customPropertyValue);
+
+        HashMap<String, String > customMap = new HashMap<>();
+        customMap.put(customPropertyKey, customPropertyValue);
+        User expectedUser = User.newBuilder().email(email).country(country).custom(customMap).build(
+            evaluationContext.getTargetingKey());
```

Comment:
It is used as the targeting key a few lines above: 
```
evaluationContext.setTargetingKey(userId);
```

Reply:
Yes, and we know the evaluationContext do nothing with the Id. But if we handle like a black-box it could mean you got a different value back when you call the getTargetingKey. The nice thing would be to use the userId here.

Reply:
transform() use it to build the user:
```
userBuilder.build(ctx.getTargetingKey());
```

and the assert is checking that userId as expected:
```
assertEquals(expectedUser.toString(), transformedUser.toString());
```

Am I missing something ? can you be more specific ?

Reply:
My point is that in the feature, the `evaluationContext.getTargetingKey` will return not "a" (this is the `userId` value in this test case) but "evaluationContextID:a"  because someone modified the code this way. Your test will run successfully. If the `expectedUser` is built with the original variable `userId`, then the test will fail and show that there is a problem.

This test should prove, the `transform` will create the same ConfigCat User from evaluationContext, what you would create from the same input values.

Reply:
fixed now